apps/yapms/src/routes/app/[country]/[map]/[year]/initialize/LoadRegions.ts

```diff
@@ -111,17 +112,26 @@ function loadRegions(node: HTMLDivElement): void {
 		const candidatesStringified = node.querySelector('svg')?.getAttribute('candidates'); //This doesn't return SVG other than the map SVG
 		const candidates = candidatesStringified != null ? JSON.parse(candidatesStringified) : null; //If candidate property not set, set candidates to null so the next check knows to use default candidates.
 		if (candidates !== null) CandidatesStore.set(CandidateStoreSchema.parse(candidates)); //If no candidates are defined in SVG, use generics defined in stores/Candidates.ts
+		const countsMap = new Map<string, number>(); //Create a map based on the data within count values, see Senate '24 for a map that implements this.
+		for (const candidate of candidates) {
+			if (candidate.count !== undefined) {
+				countsMap.set(candidate.id, candidate.count);
+			}
+		}
+		DefaultCountsStore.set(countsMap);
 	} catch (error) {
 		console.error('Error Parsing Candidate Data from Map:\n\n' + error);
 	}
 
 	//If map being loaded has the default-mode property set, change the mode.
 	const defaultModeAttribute = node.querySelector('svg')?.getAttribute('default-mode');
-	const defaultMode = ModeSchema.safeParse(defaultModeAttribute);
-	if (defaultMode.success) {
-		ModeStore.set(defaultMode.data);
-	} else {
-		console.error('Error Parsing defaultMode attribute from Map:\n\n' + defaultMode.error);
+	if (defaultModeAttribute !== null) {
+		const defaultMode = ModeSchema.safeParse(defaultModeAttribute);
+		if (defaultMode.success) {
+			ModeStore.set(defaultMode.data);
+		} else {
+			console.error('Error Parsing defaultMode attribute from Map:\n\n' + defaultMode.error);
+		}
```

Comment:
I think all of the attributes on the maps should be 100% required. That way when a map is created, it's 100% consistent and doesn't rely on the default behaviors of the website.

If we ever change the default behavior of the website, we need to look for where that default behavior change doesn't make sense, which means checking every single map without an explicit parameter. Where as if it is explicit, we have fewer situations like that.

Reply:
I will break out making all attributes required into a separate Issue/PR and leave it untouched in this one.

Reply:
Ahh Looking at the code, I guess it was never required in the first place. My bad.

But I think the original behavior of logging the error on null was fine here.